todo-app/src/App.js

```diff
@@ -1,13 +1,109 @@
 import React, { Component } from 'react';
+import Task from './Task';
+//import tasks from './task.json';
+import TaskForm from './taskForm';
+import Moment from 'moment';
 
-class App extends Component {
+export default class App extends Component {
+
+  constructor(props) {
+    super(props);
+	let tasks = [];
+	if (localStorage.tasks != null) { 
+		tasks = JSON.parse(localStorage.tasks);
+	}
+    this.state = {
+      tasksArr: tasks,
+	  editingTaskID: null
+    };
+
+    this.handleDoneChange = this.handleDoneChange.bind(this);
+  }
+  
+  handleDoneChange(doneValue, taskId) {
+    const updatedtasks = this.state.tasksArr.map(task => {
+      if (taskId === task.id) {
+        return Object.assign({}, task, {done: doneValue});
+      }
+      return task;
+    });
+    this.setState({tasksArr: updatedtasks});
+	localStorage.tasks = JSON.stringify(updatedtasks);
+  }
+  
+  handleTaskAdd = taskText => {
+	const ids = this.state.tasksArr.map(task => task.id);
+    const newTask =  {
+      id:     ids.length === 0 ? 1 : Math.max(...ids) + 1,
+      description: taskText,
+      deadline:   Moment().format('YYYY-MM-DD'),
+      done:   false
+    };
+	const newTaskArr = [newTask, ...this.state.tasksArr];
+	this.setState({tasksArr:newTaskArr});
+    localStorage.tasks = JSON.stringify(newTaskArr);
+  };
+  
+  handleTaskDelete = taskId => {
+	const newTaskArr = this.state.tasksArr.filter(task => taskId !== task.id);
+	this.setState({tasksArr:newTaskArr});
+    localStorage.tasks = JSON.stringify(newTaskArr);
+  };
+  
+	handleTaskEdit = taskId => {
+		this.setState({editingTaskID: taskId});
+	};
+	
+	handleTaskCancelEdit = () => {
+		this.setState({editingTaskID: null});
+	};
+	
+	handleTaskSave = (taskId, taskDescription) => {
+		const updatedtasks = this.state.tasksArr.map(task => {
+		  if (taskId === task.id) {
+			return Object.assign({}, task, {description: taskDescription});
+		  }
+		  return task;
+		});
+		this.setState({tasksArr: updatedtasks , editingTaskID: null});
+		localStorage.tasks = JSON.stringify(updatedtasks);
+	};
+  
   render() {
+	  
+	if (this.state.tasksArr.length === 0) {
```

Comment:
Your `if` clause wrapping this JSX is way bigger than it needs to be. In reality you're only switching over your `No items...` vs the `<ul className='task-list'>`. No need to repeat your headers and `TaskForm` component.

I realise you were probably trying to make it clearer exactly what is being returned in what cases, but this would actually get harder to read if the complexity increased any more. Of course this is somewhat a matter of preference, but you can also use the `&&` syntax or a ternary inside a single `return`.

If you don't like that, you can take after @joostlubach's examples with something like:

    return (
      <div>
        <h1>Todo list</h1>
        <h3>Enter description:</h3>
        <TaskForm
          onTaskAdd={this.handleTaskAdd}
        />
        {this.renderTaskList()}
      </div>
    );

Wherein your two view conditions are contained inside the `renderTaskList` function.